src/plugins/kibana_utils/public/state_management/url/kbn_url_tracker.ts

```diff
@@ -101,6 +102,14 @@ export function createKbnUrlTracker({
    * Storage object to use to persist currently active url. If this isn't provided, the browser wide session storage instance will be used.
    */
   storage?: Storage;
+  /**
+   * Checks if pathname belongs to current app. It's used in history listener to define whether it's necessary to set pathname as active url or not.
+   * By default the app name compares to the first part of pathname. This function is required for more complex cases.
+   *
+   * @param {string} pathname A location's pathname which comes to history listener
+   * @param {string} currentAppName An app name which is extracted from defaultSubUrl
+   */
+  isUrlBelongsToApp?: (pathname: string, currentAppName: string) => boolean;
```

Comment:
The current app name is known to the caller of kbn url tracker, so I don't see a reason why we have to pass it in as a second parameter again.

nit: Can we rename the parameter to `shouldTrackUrlUpdate`?

Reply:
Done.